src/App.jsx

```diff
@@ -14,33 +16,145 @@ export const goodsFromServer = [
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const SORT_ALPHABETICALLY = 'alphabetically';
+  const SORT_BY_LENGTH = 'sort by length';
+  const REVERSE = 'reverse';
+  const RESET = 'reset';
+  const [sort, setSort] = useState('');
+  const [reversePressed, setReversePressed] = useState(false);
+  const [alphPressed, setAlphPressed] = useState(false);
+  const [lengthPressed, setLengthPressed] = useState(false);
+  const [sortedGoods, setSortedGoods] = useState([...goodsFromServer]);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const handleSort = type => {
+    let newGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (reversePressed) {
+      switch (type) {
+        case SORT_ALPHABETICALLY:
+          newGoods.sort((good1, good2) => good2.localeCompare(good1));
+          setLengthPressed(false);
+          break;
+        case SORT_BY_LENGTH:
+          newGoods = [...sortedGoods].sort(
+            (good1, good2) => good2.length - good1.length,
+          );
+          setAlphPressed(false);
+
+          break;
+        case REVERSE:
+          newGoods = [...sortedGoods].reverse();
+          break;
+        case RESET:
+          newGoods = [...goodsFromServer];
+          setReversePressed(false);
+          setAlphPressed(false);
+          setLengthPressed(false);
+          break;
+        default:
+          break;
+      }
+    } else {
+      switch (type) {
+        case SORT_ALPHABETICALLY:
+          newGoods.sort((good1, good2) => good1.localeCompare(good2));
+          setLengthPressed(false);
+          break;
+        case SORT_BY_LENGTH:
+          newGoods.sort((good1, good2) => good1.length - good2.length);
+          setAlphPressed(false);
+
+          break;
+        case REVERSE:
+          newGoods = [...sortedGoods].reverse();
+          break;
+        case RESET:
+          newGoods = [...goodsFromServer];
+          setReversePressed(false);
+          setAlphPressed(false);
+          setLengthPressed(false);
+          break;
+        default:
+          break;
+      }
+    }
+
+    setSortedGoods(newGoods);
+    setSort(type);
+  };
 
-      <button type="button" className="button is-danger is-light">
+  const renderResetButton = () => {
+    return (
+      <button
+        type="button"
+        className="button is-danger"
+        onClick={() => {
+          handleSort(RESET);
+        }}
+      >
         Reset
       </button>
-    </div>
+    );
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': alphPressed !== true,
+            "NOT_ACTIVE_CLASS": sort !== SORT_ALPHABETICALLY,
```

Comment:
The class `NOT_ACTIVE_CLASS` is used here but is not defined or imported. This will cause an error. Please ensure that this class is defined or imported correctly.